AGENTS.md

```diff
@@ -0,0 +1,125 @@
+[addon-framework](https://github.com/open-cluster-management-io/addon-framework) and runs on the
+hub cluster to manage observability components on spoke clusters.
+
+**Tech Stack**: Go 1.24, Kubernetes controller-runtime, OCM addon-framework
+
+## Architecture Overview
+
+### Core Components
+
+**Addon Manager** (`main.go`) - Hub-side controller managing addon lifecycle across spoke clusters
+
+**Key Packages**:
+- `internal/addon/` - Addon framework integration
+- `internal/logging/` - ClusterLogForwarder manifest generation (cluster-logging-operator)
+- `internal/tracing/` - OpenTelemetryCollector and Instrumentation manifests (RHBOTO)
+- `internal/metrics/` - PrometheusAgent and ScrapeConfig manifests (prometheus-operator)
+- `internal/controllers/` - Kubernetes reconciliation controllers
+- `internal/perses/` - Dashboard definitions
+- `internal/coo/` - cluster-logging-operator utilities
+- `internal/analytics/` - Analytics reporting
+
+### How It Works
+
+1. User configures observability capabilities via `AddOnDeploymentConfig` on hub cluster
+2. Addon manager builds manifests based on configuration stanzas
+3. Manifests deployed to spoke clusters via `ManifestWork` resources
+4. Operators install on spoke clusters and forward signals to configured destinations
+
+### Three Observability Signals
+
+- **Metrics**: PrometheusAgent scrapes and forwards metrics
+- **Logs**: ClusterLogForwarder collects and forwards logs
+- **Traces**: OpenTelemetryCollector receives and forwards traces
+
+## Project Structure
+
+```
+.
+├── main.go                    # Addon manager entry point
+├── internal/
+│   ├── addon/                 # Addon framework integration
+│   ├── controllers/           # Kubernetes controllers
+│   ├── logging/               # ClusterLogForwarder logic
+│   ├── tracing/               # OpenTelemetry logic
+│   ├── metrics/               # Prometheus metrics logic
+│   ├── perses/                # Dashboard definitions
+│   ├── analytics/             # Analytics and reporting
+│   └── coo/                   # Cluster-logging-operator utils
+├── deploy/                    # Kustomize deployment manifests
+├── hack/                      # Development tools and test resources
+└── .bingo/                    # Tool dependency management
+```
+
+## Development Workflow
+
+### Quick Start
+
+```bash
+# Build and test
+make addon          # Build binary
+make test           # Run unit tests
+make fmt            # Format code
+make lint           # Run linters
+
+# Deploy to development cluster
+export REGISTRY_BASE=quay.io/YOUR_QUAY_ID
+make oci            # Build and push image
+make addon-deploy   # Deploy to hub cluster
+```
+
+### Iterative Development
+
+After making code changes:
+
+```bash
+make oci
+oc -n open-cluster-management-observability delete pod -l app=multicluster-observability-addon-manager
+```
+
+Additional useful targets: `make install-crds`, `make download-crds`, `make update-metrics-crds`
```

Comment:
Consider adding these, if you already know what usually gets wrong and you need further instruct:
- "Common Mistakes to Avoid" section
- "NEVER do X" warnings (eg. we have a specific way of running tests we put into this doc)